3.11/indexing-inverted.md

```diff
@@ -0,0 +1,309 @@
+db.<collection>.ensureIndex({ type: "inverted", fields: [ "arr", "arr.name" ], searchField: true, trackListPositions: true });
+```
+
+### Storing additional values in indexes
+
+Inverted indexes allow you to store additional attributes in the index that
+can be used to satisfy projections of the document. They cannot be used for
+index lookups or for sorting, but for projections only. They allow inverted
+indexes to fully cover more queries and avoid extra document lookups. This can
+have a great positive effect on index scan performance if the number of scanned
+index entries is large.
+
+You can set the `storedValues` option and specify the additional attributes as
+an array of attribute paths when creating a new inverted index:
+
+```js
+db.<collection>.ensureIndex({
+  type: "inverted",
+  fields: ["value1"],
+  storedValues: ["value2"]
+});
+```
+
+This indexes the `value1` attribute in the traditional sense, so that the index 
+can be used for looking up by `value1` or for sorting by `value1`. The index also
+supports projections on `value1` as usual.
+
+In addition, due to `storedValues` being used here, the index can now also 
+supply the values for the `value2` attribute for projections without having to
+look up the full document. Non-existing attributes are stored as `null` values.
+
+You cannot specify the same attribute path in both, the `fields` and the
+`storedValues` option. If there is an overlap, the index creation aborts
+with an error message.
+
+### Additional configuration options
+
+See the full list of options in the [HTTP API](http/indexes-inverted.html)
+documentation.
+
+### Restrictions
+
+- You cannot index the same field twice in a single inverted index. This includes
+  indexing the same field with and without array expansion (e.g.
+  `fields: ["arr", "arr[*]"]`).
+- Every field can only be processed by a single Analyzer per index. The benefit
+  is that you don't need to specify the Analyzers in queries with the `ANALYZER()`
+  function, they can be inferred from the index definition.
+
+If you plan on using inverted indexes in Search Alias Views, also consider the
+following restrictions:
+
+- All inverted indexes you add to a single `search-alias` View need to have
+  matching `primarySort` and `storedValues` settings.
+- If inverted indexes of different collections index fields with the same name
+  (attribute path), these fields need to have matching `analyzer` and
+  `searchField` settings.
+
+## Utilizing inverted indexes in queries
+
+Unlike other index types in ArangoDB, inverted indexes are not utilized
+automatically, even if they are eligible for a query. The reason is that AQL
+queries may produce different results with and without an inverted index,
+because of differences in how `FILTER` operations work for inverted indexes,
+the eventual-consistent nature of this index type, and for backward compatibility.
```

Comment:
It's also for backward compatibility because there is a mismatch between the normal FILTER semantics (no index or persistent index) vs. FILTER with an inverted index. Automatically picking an inverted index could return different query results, which is not what we want.